openexr fails to build on Mac OS X since e5e31593cebf0322fff2badcb8f85eb2cd40da74

**Describe the bug**
build of OpenEXR fails to find -lImath2_4

**To Reproduce**

```
$ nix-build . -A openexr
```

**Expected behavior**

Successful build.

**Screenshots**

```
clang++ -DHAVE_CONFIG_H -I. -I../config  -D_THREAD_SAFE -I/nix/store/yzdk3am5jpm0g87mzm513jzj1w4mp12v-ilmbase-2.4.1-dev/include -I/nix/store/yzdk3am5jpm0g87mzm513jzj1w4mp12v-ilmbase-2.4.1-dev/include/OpenEXR -D_THREAD_SAFE -I.. -I../config  -I/nix/store/wkz4cs14c5dsfi86q3zw8nayfii1lckr-zlib-1.2.11-dev/include  -pipe -g -O2 -std=c++14 -D_THREAD_SAFE  -c -o dwaLookups.o dwaLookups.cpp
/nix/store/pd2xsac3kz5ziskb5x90cm9jcjk2xcc6-bash-4.4-p23/bin/bash ../libtool  --tag=CXX   --mode=link clang++ -pipe -g -O2 -std=c++14 -D_THREAD_SAFE    -o dwaLookups dwaLookups.o -L/nix/store/gjj7ka209rfdr4fsryjfx16if4bk74qa-ilmbase-2.4.1/lib -L/nix/store/gjj7ka209rfdr4fsryjfx16if4bk74qa-ilmbase-2.4.1/lib -lImath-2_4 -lHalf-2_4 -lIex-2_4 -lIexMath-2_4 -lIlmThread-2_4 -D_THREAD_SAFE
libtool: link: clang++ -pipe -g -O2 -std=c++14 -D_THREAD_SAFE -o dwaLookups dwaLookups.o -D_THREAD_SAFE  -L/nix/store/gjj7ka209rfdr4fsryjfx16if4bk74qa-ilmbase-2.4.1/lib -lImath-2_4 -lHalf-2_4 -lIex-2_4 -lIexMath-2_4 -lIlmThread-2_4
ld: library not found for -lImath-2_4
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:762: dwaLookups] Error 1
make[1]: *** Waiting for unfinished jobs....
/nix/store/pd2xsac3kz5ziskb5x90cm9jcjk2xcc6-bash-4.4-p23/bin/bash ../libtool  --tag=CXX   --mode=link clang++ -pipe -g -O2 -std=c++14 -D_THREAD_SAFE    -o b44ExpLogTable b44ExpLogTable.o -L/nix/store/gjj7ka209rfdr4fsryjfx16if4bk74qa-ilmbase-2.4.1/lib -L/nix/store/gjj7ka209rfdr4fsryjfx16if4bk74qa-ilmbase-2.4.1/lib -lImath-2_4 -lHalf-2_4 -lIex-2_4 -lIexMath-2_4 -lIlmThread-2_4 -D_THREAD_SAFE
libtool: link: clang++ -pipe -g -O2 -std=c++14 -D_THREAD_SAFE -o b44ExpLogTable b44ExpLogTable.o -D_THREAD_SAFE  -L/nix/store/gjj7ka209rfdr4fsryjfx16if4bk74qa-ilmbase-2.4.1/lib -lImath-2_4 -lHalf-2_4 -lIex-2_4 -lIexMath-2_4 -lIlmThread-2_4
ld: library not found for -lImath-2_4
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:758: b44ExpLogTable] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-openexr-2.4.1.drv-0/source/OpenEXR/IlmImf'
make: *** [Makefile:502: all-recursive] Error 1
builder for '/nix/store/3qp40w14j9x0rf9kh58j6mxah4829mz5-openexr-2.4.1.drv' failed with exit code 2
error: build of '/nix/store/3qp40w14j9x0rf9kh58j6mxah4829mz5-openexr-2.4.1.drv' failed
```

**Additional context**
Add any other context about the problem here.

**Notify maintainers**

@lovesegfault 

**Metadata**

 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.2.0, macOS 10.15.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"nixpkgs-20.09pre221715.5f814bf9cc4"`
 - channels(jfelice): `"darwin"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

Maintainer information:
```yaml
attribute:
- openexr
- ilmbase
```

--- a/pkgs/development/libraries/ilmbase/default.nix
+++ b/pkgs/development/libraries/ilmbase/default.nix
@@ -1,9 +1,13 @@
-{ stdenv, lib, buildPackages, automake, autoconf, libtool, which,
-  fetchpatch, openexr }:
+{ stdenv
+, buildPackages
+, cmake
+, libtool
+, openexr
+}:
 
 stdenv.mkDerivation rec {
   pname = "ilmbase";
-  version = lib.getVersion openexr;
+  version = stdenv.lib.getVersion openexr;
 
   # the project no longer provides separate tarballs. We may even want to merge
   # the ilmbase package into openexr in the future.
@@ -13,19 +17,10 @@ stdenv.mkDerivation rec {
 
   outputs = [ "out" "dev" ];
 
-  preConfigure = ''
-    patchShebangs ./bootstrap
-    ./bootstrap
-  '';
-
+  nativeBuildInputs = [ cmake libtool ];
   depsBuildBuild = [ buildPackages.stdenv.cc ];
-  nativeBuildInputs = [ automake autoconf libtool which ];
 
-  NIX_CFLAGS_LINK = "-pthread";
-
-  patches = [
-    ./cross.patch
-  ];
+  patches = [ ./cross.patch ];
 
   # fails 1 out of 1 tests with
   # "lt-ImathTest: testBoxAlgo.cpp:892: void {anonymous}::boxMatrixTransform(): Assertion `b21 == b2' failed"

--- a/pkgs/development/libraries/openexr/default.nix
+++ b/pkgs/development/libraries/openexr/default.nix
@@ -1,11 +1,13 @@
-{ lib, stdenv, buildPackages, fetchFromGitHub, autoconf, automake, libtool, pkgconfig,
-  zlib, ilmbase, fetchpatch }:
-
-let
-  # Doesn't really do anything when not crosscompiling
-  emulator = stdenv.hostPlatform.emulator buildPackages;
-in
-
+{ lib
+, stdenv
+, buildPackages
+, fetchFromGitHub
+, zlib
+, ilmbase
+, fetchpatch 
+, cmake
+, libtool
+}:
 stdenv.mkDerivation rec {
   pname = "openexr";
   version = "2.4.1";
@@ -17,36 +19,11 @@ stdenv.mkDerivation rec {
     sha256 = "020gyl8zv83ag6gbcchmqiyx9rh2jca7j8n52zx1gk4rck7kwc01";
   };
 
-  sourceRoot = "source/OpenEXR";
-
   outputs = [ "bin" "dev" "out" "doc" ];
-
-  # Needed because there are some generated sources. Solution: just run them under QEMU.
-  postPatch = ''
-    for file in b44ExpLogTable dwaLookups
-    do
-      # Ecape for both sh and Automake
-      emu=${lib.escapeShellArg (lib.replaceStrings ["$"] ["$$"] emulator)}
-      before="./$file > $file.h"
-      after="$emu $before"
-      substituteInPlace IlmImf/Makefile.am \
-        --replace "$before" "$after"
-    done
-
-    # Make sure the patch succeeded
-    [[ $(grep "$emu" IlmImf/Makefile.am | wc -l) = 2 ]]
-  '';
-
-  preConfigure = ''
-    patchShebangs ./bootstrap
-    ./bootstrap
-  '';
-
-  nativeBuildInputs = [ pkgconfig autoconf automake libtool ];
+  nativeBuildInputs = [ cmake libtool ];
   propagatedBuildInputs = [ ilmbase zlib ];
 
   enableParallelBuilding = true;
-  doCheck = false; # fails 1 of 1 tests
 
   meta = with stdenv.lib; {
     description = "A high dynamic-range (HDR) image file format";

